packages/boxel-ui/test-app/tests/integration/components/resizable-panel-group-horizontal-test.gts

```diff
@@ -0,0 +1,310 @@
+      new PanelProperties(
+        { defaultLengthFraction: 0.6 },
+        {
+          showResizeHandle: true,
+          outerContainerStyle:
+            'border: 1px solid red; height: 100%; overflow-y:auto',
+        },
+      ),
+      new PanelProperties(
+        { defaultLengthFraction: 0.4, minLengthPx: 50 },
+        {
+          outerContainerStyle: 'border: 1px solid red; height: 100%',
+        },
+      ),
+    ];
+  });
+
+  async function renderHorizontalResizablePanelGroup(
+    renderController: RenderController,
+  ) {
+    await render(<template>
+      {{! template-lint-disable no-inline-styles }}
+      <div id='test-container' style={{renderController.containerStyle}}>
+        <ResizablePanelGroup
+          @orientation='horizontal'
+          @reverseCollapse={{true}}
+          as |ResizablePanel ResizeHandle|
+        >
+          {{#each renderController.panels as |panel index|}}
+
+            <ResizablePanel
+              @defaultLengthFraction={{panel.defaultLengthFraction}}
+              @lengthPx={{panel.lengthPx}}
+              @minLengthPx={{panel.minLengthPx}}
+              @isHidden={{panel.isHidden}}
+            >
+              {{#if (eq index 1)}}
+                <div
+                  class='panel-{{index}}-content'
+                  style={{panel.outerContainerStyle}}
+                >
+                  <div>
+                    Panel
+                    {{index}}
+                  </div>
+                </div>
+              {{else}}
+                <div
+                  class='panel-{{index}}-content'
+                  style={{panel.outerContainerStyle}}
+                >
+                  Panel 2
+                </div>
+              {{/if}}
+            </ResizablePanel>
+            {{#if panel.showResizeHandle}}
+              <ResizeHandle />
+            {{/if}}
+          {{/each}}
+        </ResizablePanelGroup>
+      </div>
+    </template>);
+    await sleep(100); // let didResizeModifier run
+  }
+
+  test<MyTestContext>('it can lay out panels horizontally', async function (assert) {
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 100px; width: 318px;';
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    assert.hasNumericStyle('.panel-0-content', 'width', 300 * 0.6, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 300 * 0.4, 1);
+  });
+
+  test<MyTestContext>('it can lay out panels horizontally (length specified)', async function (assert) {
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 518px; border: 1px solid green';
+    this.renderController.panels[0].lengthPx = 355;
+    this.renderController.panels[1].lengthPx = 143;
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    assert.hasNumericStyle('.panel-0-content', 'width', 355, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 143, 1);
+  });
```

Comment:
like this test has 355+143=498. So the surplus is like 20. Then im a bit cofused.

Reply:
I found it’s because of the 1px border, which I’ve removed in f202bab, along with other seemingly-extraneous styles.